examples/mongocxx/mongodb.com/documentation_examples.cpp

```diff
@@ -57,6 +61,119 @@ void check_has_no_field(const T& document, const char* field, int example_no) {
+        } else {
+            throw std::logic_error("invalid server API version");
+        }
+    }
+    return opts;
+}
+
+std::string getenv_or_fail(const char* s) {
+    const char* env = std::getenv(s);
+    if (!env) {
+        throw std::runtime_error("missing a required environment variable: " + std::string(s));
+    }
+    return std::string(env);
+}
+
+// Returns a document with credentials for KMS providers.
+// If include_external is true, all KMS providers are set.
+// If include_external is false, only the local provider is set.
+bsoncxx::document::value _make_kms_doc(bool include_external = true) {
+    using bsoncxx::builder::basic::sub_document;
+    using bsoncxx::builder::stream::close_array;
+    using bsoncxx::builder::stream::close_document;
+    using bsoncxx::builder::stream::document;
+    using bsoncxx::builder::stream::finalize;
+    using bsoncxx::builder::stream::open_array;
+    using bsoncxx::builder::stream::open_document;
+
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+
+    const auto kLocalMasterKey =
+        "\x32\x78\x34\x34\x2b\x78\x64\x75\x54\x61\x42\x42\x6b\x59\x31\x36\x45\x72"
+        "\x35\x44\x75\x41\x44\x61\x67\x68\x76\x53\x34\x76\x77\x64\x6b\x67\x38\x74"
+        "\x70\x50\x70\x33\x74\x7a\x36\x67\x56\x30\x31\x41\x31\x43\x77\x62\x44\x39"
+        "\x69\x74\x51\x32\x48\x46\x44\x67\x50\x57\x4f\x70\x38\x65\x4d\x61\x43\x31"
+        "\x4f\x69\x37\x36\x36\x4a\x7a\x58\x5a\x42\x64\x42\x64\x62\x64\x4d\x75\x72"
+        "\x64\x6f\x6e\x4a\x31\x64";
+
+    auto kms_doc = bsoncxx::builder::basic::document{};
+
+    if (include_external) {
+        kms_doc.append(kvp("aws", [&](sub_document subdoc) {
+            subdoc.append(
+                kvp("secretAccessKey", getenv_or_fail("MONGOCXX_TEST_AWS_SECRET_ACCESS_KEY")));
+            subdoc.append(kvp("accessKeyId", getenv_or_fail("MONGOCXX_TEST_AWS_ACCESS_KEY_ID")));
+        }));
+
+        kms_doc.append(kvp("azure", [&](sub_document subdoc) {
+            subdoc.append(kvp("tenantId", getenv_or_fail("MONGOCXX_TEST_AZURE_TENANT_ID")));
+            subdoc.append(kvp("clientId", getenv_or_fail("MONGOCXX_TEST_AZURE_CLIENT_ID")));
+            subdoc.append(kvp("clientSecret", getenv_or_fail("MONGOCXX_TEST_AZURE_CLIENT_SECRET")));
+        }));
+
+        kms_doc.append(kvp("gcp", [&](sub_document subdoc) {
+            subdoc.append(kvp("email", getenv_or_fail("MONGOCXX_TEST_GCP_EMAIL")));
+            subdoc.append(kvp("privateKey", getenv_or_fail("MONGOCXX_TEST_GCP_PRIVATEKEY")));
+        }));
+
+        kms_doc.append(kvp("kmip", [&](sub_document subdoc) {
+            subdoc.append(kvp("endpoint", "localhost:5698"));
+        }));
+    }
+
+    char key_storage[96];
+    memcpy(&(key_storage[0]), kLocalMasterKey, 96);
+
+    bsoncxx::types::b_binary local_master_key{
+        bsoncxx::binary_sub_type::k_binary, 96, (const uint8_t*)&key_storage};
```

Comment:
```suggestion
    bsoncxx::types::b_binary local_master_key{
        bsoncxx::binary_sub_type::k_binary, 96, kLocalMasterKey};
```
Enabled by change to `kLocalMasterKey` suggested above.